docs/topics/native/native-ios-integration.md

```diff
@@ -244,6 +245,27 @@ It requires two GC cycles to collect these four objects because deinitialization
 after the GC cycle. The limitation stems from `deinit`, which can call arbitrary code, including the Kotlin code that
 cannot be run during the GC pause.
 
+### Retain cycles
+
+In a _retain cycle_, a number of objects refer each other using strong references cyclically:
+
+![Retain cycles](native-retain-cycle.png){height=200}
+
+Kotlin's tracing GC and Objective-C's ARC handle retain cycles differently. When objects become unreachable, tracing GC
+can properly reclaim such cycles, while Objective-C's ARC cannot. Therefore, retain cycles of Kotlin objects can be reclaimed,
+while [retain cycles of Swift/Objective-C objects cannot](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/automaticreferencecounting/#Strong-Reference-Cycles-Between-Class-Instances).
+
+Consider the case when a retain cycle contains both Objective-C and Kotlin objects:
+
+![Retain cycles with Objective-C and Kotlin objects](native-objc-kotlin-retain-cycles.png){height=150}
+
+This involves combining Kotlin's and Objective-C's memory management models that cannot handle (reclaim) retain cycles
+together. That means that if at least one Objective-C object is present, the retain cycle of a whole graph of objects
```

Comment:
```suggestion
together. Meaning that if at least one Objective-C object is present, the retain cycle of a whole graph of objects
```

Reply:
Or consider even removing that part and starting the sentence with "If". To avoid repeating "that" so closely.

Reply:
I removed the second 'that' to avoid repetition, I think that works too